### DISCUSSION TITLE
Incident Notes

### DISCUSSION THREAD

**Discussion by ithakaa** [henrygd/beszel#241] [Ideas]
I’d like to propose a feature enhancement for our monitoring system that could improve our ability to analyze and contextualize historical data. The idea is to implement an "Incident Notes" section directly within the monitoring interface, allowing us to annotate graphs and other metrics with contextual notes during or after an incident.

Key Features:
- **Graph Annotations**: Ability to tag specific points or periods on graphs with incident notes to explain a spike, dip, or outage.
- **Timestamped Notes**: Each note would automatically include a timestamp to match with the exact time of the anomaly.
- **Collaborative Access**: Notes could be accessible and editable by team members involved in the incident response to capture a complete view of the event.
- **Searchable History**: A searchable notes archive could be valuable for troubleshooting, enabling us to quickly identify patterns or recurring issues by viewing past incidents.

This addition could serve as a reference point for anyone reviewing metrics, particularly useful for new team members or when conducting post-mortems. Please let me know if we could discuss this further—I’d be happy to go over specifics and brainstorm additional functionalities.

**Comment by henrygd** [comment]
That could be a good feature.

We may be limited by the chart library as it doesn't seem to support annotations natively: https://github.com/recharts/recharts/issues/1678

Maybe dedicated pages for incidents at `/system/name/incidents` and incident detail at `/system/name/incidents/timestamp` where it can be modified?

If you or anyone else has further details as far as specific functionality / UI needed for this, let me know.

I'll try to implement it at some point if we have a clear picture of what it will look like.
